## 88% of end-end tests have passed
```
Total executed tests: 51
Failed tests: 6
Expected to fail tests: 0
Passed tests: 45
```
```
IDs of failed tests: 727231,702894,703496,702948,739307,703297 
```
<h3>Failed tests (6)</h3><details><summary>Click to expand</summary><br/><li><a href="https://ci.status.im/job/status-mobile/job/e2e/job/status-app-prs-rerun/parambuild/?BRANCH_NAME=fix/hidden-community&PR_ID=21238&APK_NAME=21238.apk&TR_CASE_IDS=727231,702894,703496,702948,739307,703297">Rerun failed tests</a></li><br/><h4>Class TestCommunityMultipleDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3670719">test_community_contact_block_unblock_offline</a>, id: 702894 </b></td></tr><tr><td><p><blockquote><div>Device 1: Click until `ChatMessageInput` by `accessibility id`: `chat-message-input` will be presented </div><div>Device 1: Looking for a message by text: Hurray! unblocked </div></blockquote></p>

```
critical/chats/test_public_chat_browsing.py:733: in test_community_contact_block_unblock_offline
    self.errors.verify_no_errors()
base_test_case.py:191: in verify_no_errors
    pytest.fail('\n '.join([self.errors.pop(0) for _ in range(len(self.errors))]))
 Hurray! unblocked was not received in public chat after user unblock!
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/2f469dd578f54cf88294a1017ac04e5d?auth=36993566e30ff17b053e3572c47d5990#419">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/9c0113876dee43818705e5ce5c639457?auth=b4c3ac6fd45915ece15c33879368f3b6#401">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestGroupChatMultipleDeviceMergedNewUI:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3670728">test_group_chat_send_image_save_and_share</a>, id: 703297 </b></td></tr><tr><td><p><blockquote><div>Device 3: Find `ChatsTab` by `accessibility id`: `chats-stack-tab` </div><div>Device 3: Tap on found: ChatsTab </div></blockquote></p>

```
critical/chats/test_group_chat.py:297: in test_group_chat_send_image_save_and_share
    self.errors.verify_no_errors()
base_test_case.py:191: in verify_no_errors
    pytest.fail('\n '.join([self.errors.pop(0) for _ in range(len(self.errors))]))
 Message about saving a photo is not shown for member_2.
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/580b57c1662a45289adc297d459655ff?auth=d47713a0e1915a2c27cbd35b2a9162ec#276">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/7de2a1ac49114d558559dcc36f26cd39?auth=8cd9ce39eeee3fb6a045246bce71464d#179">Steps, video, logs</a></li></ul></p><p>Device 3:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/66ba3a22093e4f80848c69f3733bf904?auth=c404ef16d6d5fc94eb62c2e2eaeb2446#149">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestWalletOneDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3670739">test_wallet_add_remove_regular_account</a>, id: 727231 </b></td></tr><tr><td><p><blockquote><div>Device 1: Wait for text element `Text` to be equal to `New Account` </div><div>Device 1: Find `Text` by `xpath`: `//*[@content-desc='link-to-profile']/preceding-sibling::android.widget.TextView` </div></blockquote></p>

```
critical/test_wallet.py:227: in test_wallet_add_remove_regular_account
    self.home_view.account_name_text.wait_for_element_text(text=new_account_name, wait_time=3)
../views/base_element.py:181: in wait_for_element_text
    element_text = self.find_element().text.strip()
../views/base_element.py:84: in find_element
    raise exception
../views/base_element.py:77: in find_element
    return self.driver.find_element(self.by, self.locator)
../../../../status-app-prs-rerun@tmp/venv/lib/python3.10/site-packages/appium/webdriver/webdriver.py:409: in find_element
    return self.execute(RemoteCommand.FIND_ELEMENT, {'using': by, 'value': value})['value']
../../../../status-app-prs-rerun@tmp/venv/lib/python3.10/site-packages/selenium/webdriver/remote/webdriver.py:345: in execute
    self.error_handler.check_response(response)
../../../../status-app-prs-rerun@tmp/venv/lib/python3.10/site-packages/appium/webdriver/errorhandler.py:122: in check_response
    raise exception_class(msg=message, stacktrace=format_stacktrace(stacktrace))
 An unknown server-side error occurred while processing the command. Original error: Could not proxy command to the remote server. Original error: socket hang up
E   Stacktrace:
E   UnknownError: An unknown server-side error occurred while processing the command. Original error: Could not proxy command to the remote server. Original error: socket hang up
E       at UIA2Proxy.command (/mnt/sauce/appium/appium-v2.0.0/packages/base-driver/lib/jsonwp-proxy/proxy.js:355:13)
E       at processTicksAndRejections (node:internal/process/task_queues:95:5)
E       at AndroidUiautomator2Driver.helpers.doFindElementOrEls (/mnt/sauce/appium/appium-v2.0.0/appium_home/linux/node_modules/appium-uiautomator2-driver/lib/commands/find.js:30:10)
E       at doFind (/mnt/sauce/appium/appium-v2.0.0/appium_home/linux/node_modules/appium-uiautomator2-driver/node_modules/appium-android-driver/lib/commands/find.js:36:17)
E       at wrappedCondFn (/mnt/sauce/appium/appium-v2.0.0/packages/base-driver/lib/basedriver/commands/timeout.ts:137:14)
E       at spin (/mnt/sauce/appium/appium-v2.0.0/node_modules/asyncbox/lib/asyncbox.js:221:20)
E       at waitForCondition (/mnt/sauce/appium/appium-v2.0.0/node_modules/asyncbox/lib/asyncbox.js:238:10)
E       at AndroidUiautomator2Driver.implicitWaitForCondition (/mnt/sauce/appium/appium-v2.0.0/packages/base-driver/lib/basedriver/commands/timeout.ts:139:12)
E       at AndroidUiautomator2Driver.findElOrEls (/mnt/sauce/appium/appium-v2.0.0/appium_home/linux/node_modules/appium-uiautomator2-driver/node_modules/appium-android-driver/lib/commands/find.js:60:5)
E       at AndroidUiautomator2Driver.findElOrElsWithProcessing (/mnt/sauce/appium/appium-v2.0.0/packages/base-driver/lib/basedriver/commands/find.ts:60:12)
E       at AndroidUiautomator2Driver.findElement (/mnt/sauce/appium/appium-v2.0.0/packages/base-driver/lib/basedriver/commands/find.ts:75:12)
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/d940b7f10c8d4e928791a2b356ff5987?auth=95b1479464da4eb419a4b61997f354e5#39">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestCommunityMultipleDeviceMergedTwo:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3670721">test_community_hashtag_links_to_community_channels</a>, id: 702948 </b></td></tr><tr><td><p><blockquote><div>Device 2: Find `Text` by `xpath`: `//android.view.ViewGroup[@content-desc='chat-item']//android.widget.TextView[contains(@text,'#cats')]` </div><div>Device 2: Looking for a message by text: it is just a message text </div></blockquote></p>

```
critical/chats/test_public_chat_browsing.py:1122: in test_community_hashtag_links_to_community_channels
    self.errors.verify_no_errors()
base_test_case.py:191: in verify_no_errors
    pytest.fail('\n '.join([self.errors.pop(0) for _ in range(len(self.errors))]))
 Message in community channel is not visible for user before join
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/ac752e7299ad4a53852d6f5127ac1853?auth=101ed1010416a1688c32ff3d25a58ffa#309">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/963af9dfc87846fcb422f93bb1072c7b?auth=61dec3b0b49a08aac410b887d3a96dd0#299">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestDeepLinksOneDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3670740">test_deep_links_communities</a>, id: 739307 </b></td></tr><tr><td><p><blockquote><div>Device 1: Waiting 35s for dapp to load </div><div>Device 1: Find `Text` by `accessibility id`: `community-title` </div></blockquote></p>

```
critical/test_deep_and_universal_links.py:105: in test_deep_links_communities
    if self.community_view.community_title.text != text:
../views/base_element.py:419: in text
    text = self.find_element().text
../views/base_element.py:79: in find_element
    raise NoSuchElementException(
 Device 1: Text by accessibility id: `community-title` is not found on the screen; For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/203a395665c6464da2803d5cf442d303?auth=bae99f678497a6b631c142857610e559#195">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestOneToOneChatMultipleSharedDevicesNewUiTwo:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3670732">test_1_1_chat_mute_chat</a>, id: 703496 </b></td></tr><tr><td><p><blockquote><div>Device 1: Click until `ChatMessageInput` by `accessibility id`: `chat-message-input` will be presented </div><div>Device 1: Looking for a message by text: after unmute </div></blockquote></p>

```
critical/chats/test_1_1_public_chats.py:666: in test_1_1_chat_mute_chat
    self.errors.verify_no_errors()
base_test_case.py:191: in verify_no_errors
    pytest.fail('\n '.join([self.errors.pop(0) for _ in range(len(self.errors))]))
 Message text 'should be muted' is not shown in chat preview after mute
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/7925e95f2ee2499f8c70304dfd450127?auth=5fa9a9d673016d08aadd3ad08f88edab#110">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/1c3936beddc645a990fa9667cd68d7cd?auth=ef35f8feaa71fbe53a0250c3b6ec4b6f#156">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table></details><h3>Passed tests (45)</h3><details><summary>Click to expand</summary><br/><br/><h4>Class TestCommunityMultipleDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3670708">test_community_emoji_send_copy_paste_reply</a>, id: 702840 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/2f469dd578f54cf88294a1017ac04e5d?auth=36993566e30ff17b053e3572c47d5990#300">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/9c0113876dee43818705e5ce5c639457?auth=b4c3ac6fd45915ece15c33879368f3b6#291">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3670724">test_community_mark_all_messages_as_read</a>, id: 703086 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/2f469dd578f54cf88294a1017ac04e5d?auth=36993566e30ff17b053e3572c47d5990#544">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/9c0113876dee43818705e5ce5c639457?auth=b4c3ac6fd45915ece15c33879368f3b6#424">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3670711">test_community_links_with_previews_github_youtube_twitter_gif_send_enable</a>, id: 702844 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/2f469dd578f54cf88294a1017ac04e5d?auth=36993566e30ff17b053e3572c47d5990#324">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/9c0113876dee43818705e5ce5c639457?auth=b4c3ac6fd45915ece15c33879368f3b6#310">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3670709">test_community_unread_messages_badge</a>, id: 702841 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/2f469dd578f54cf88294a1017ac04e5d?auth=36993566e30ff17b053e3572c47d5990#389">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/9c0113876dee43818705e5ce5c639457?auth=b4c3ac6fd45915ece15c33879368f3b6#371">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>5. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3670707">test_community_message_delete</a>, id: 702839 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/2f469dd578f54cf88294a1017ac04e5d?auth=36993566e30ff17b053e3572c47d5990#206">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/9c0113876dee43818705e5ce5c639457?auth=b4c3ac6fd45915ece15c33879368f3b6#143">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>6. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3670706">test_community_message_send_check_timestamps_sender_username</a>, id: 702838 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/2f469dd578f54cf88294a1017ac04e5d?auth=36993566e30ff17b053e3572c47d5990#164">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/9c0113876dee43818705e5ce5c639457?auth=b4c3ac6fd45915ece15c33879368f3b6#114">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>7. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3670736">test_community_edit_delete_message_when_offline</a>, id: 704615 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/2f469dd578f54cf88294a1017ac04e5d?auth=36993566e30ff17b053e3572c47d5990#582">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/9c0113876dee43818705e5ce5c639457?auth=b4c3ac6fd45915ece15c33879368f3b6#453">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>8. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3670717">test_community_one_image_send_reply</a>, id: 702859 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/2f469dd578f54cf88294a1017ac04e5d?auth=36993566e30ff17b053e3572c47d5990#252">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/9c0113876dee43818705e5ce5c639457?auth=b4c3ac6fd45915ece15c33879368f3b6#235">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>9. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3670710">test_community_message_edit</a>, id: 702843 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/2f469dd578f54cf88294a1017ac04e5d?auth=36993566e30ff17b053e3572c47d5990#182">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/9c0113876dee43818705e5ce5c639457?auth=b4c3ac6fd45915ece15c33879368f3b6#131">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>10. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3670726">test_community_several_images_send_reply</a>, id: 703194 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/2f469dd578f54cf88294a1017ac04e5d?auth=36993566e30ff17b053e3572c47d5990#227">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/9c0113876dee43818705e5ce5c639457?auth=b4c3ac6fd45915ece15c33879368f3b6#172">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestGroupChatMultipleDeviceMergedNewUI:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3670727">test_group_chat_reactions</a>, id: 703202 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/580b57c1662a45289adc297d459655ff?auth=d47713a0e1915a2c27cbd35b2a9162ec#169">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/7de2a1ac49114d558559dcc36f26cd39?auth=8cd9ce39eeee3fb6a045246bce71464d#117">Steps, video, logs</a></li></ul></p><p>Device 3:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/66ba3a22093e4f80848c69f3733bf904?auth=c404ef16d6d5fc94eb62c2e2eaeb2446#94">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3670702">test_group_chat_join_send_text_messages_push</a>, id: 702807 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/580b57c1662a45289adc297d459655ff?auth=d47713a0e1915a2c27cbd35b2a9162ec#152">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/7de2a1ac49114d558559dcc36f26cd39?auth=8cd9ce39eeee3fb6a045246bce71464d#89">Steps, video, logs</a></li></ul></p><p>Device 3:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/66ba3a22093e4f80848c69f3733bf904?auth=c404ef16d6d5fc94eb62c2e2eaeb2446#86">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3670703">test_group_chat_offline_pn</a>, id: 702808 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/580b57c1662a45289adc297d459655ff?auth=d47713a0e1915a2c27cbd35b2a9162ec#316">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/7de2a1ac49114d558559dcc36f26cd39?auth=8cd9ce39eeee3fb6a045246bce71464d#214">Steps, video, logs</a></li></ul></p><p>Device 3:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/66ba3a22093e4f80848c69f3733bf904?auth=c404ef16d6d5fc94eb62c2e2eaeb2446#208">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3670692">test_group_chat_pin_messages</a>, id: 702732 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/580b57c1662a45289adc297d459655ff?auth=d47713a0e1915a2c27cbd35b2a9162ec#349">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/7de2a1ac49114d558559dcc36f26cd39?auth=8cd9ce39eeee3fb6a045246bce71464d#241">Steps, video, logs</a></li></ul></p><p>Device 3:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/66ba3a22093e4f80848c69f3733bf904?auth=c404ef16d6d5fc94eb62c2e2eaeb2446#235">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>5. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3670731">test_group_chat_mute_chat</a>, id: 703495 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/580b57c1662a45289adc297d459655ff?auth=d47713a0e1915a2c27cbd35b2a9162ec#469">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/7de2a1ac49114d558559dcc36f26cd39?auth=8cd9ce39eeee3fb6a045246bce71464d#284">Steps, video, logs</a></li></ul></p><p>Device 3:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/66ba3a22093e4f80848c69f3733bf904?auth=c404ef16d6d5fc94eb62c2e2eaeb2446#249">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestCommunityMultipleDeviceMergedTwo:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3670712">test_community_leave</a>, id: 702845 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/ac752e7299ad4a53852d6f5127ac1853?auth=101ed1010416a1688c32ff3d25a58ffa#306">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/963af9dfc87846fcb422f93bb1072c7b?auth=61dec3b0b49a08aac410b887d3a96dd0#277">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3670701">test_community_mentions_push_notification</a>, id: 702786 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/ac752e7299ad4a53852d6f5127ac1853?auth=101ed1010416a1688c32ff3d25a58ffa#141">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/963af9dfc87846fcb422f93bb1072c7b?auth=61dec3b0b49a08aac410b887d3a96dd0#117">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3670704">test_community_markdown_support</a>, id: 702809 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/ac752e7299ad4a53852d6f5127ac1853?auth=101ed1010416a1688c32ff3d25a58ffa#190">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/963af9dfc87846fcb422f93bb1072c7b?auth=61dec3b0b49a08aac410b887d3a96dd0#175">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3670734">test_community_join_when_node_owner_offline</a>, id: 703629 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/ac752e7299ad4a53852d6f5127ac1853?auth=101ed1010416a1688c32ff3d25a58ffa#504">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/963af9dfc87846fcb422f93bb1072c7b?auth=61dec3b0b49a08aac410b887d3a96dd0#479">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestActivityCenterContactRequestMultipleDevicePR:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3670715">test_activity_center_contact_request_accept_swipe_mark_all_as_read</a>, id: 702851 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/1dc8a8aaa39640bab120e5ac22bfb517?auth=6dbe1d2861d2387590fd9f50e25bad29#107">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/1eac35043abc46caa7523ec2d5c2eeaa?auth=83d3f0440e70e7fd55157c8b896a2f75#112">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3670714">test_activity_center_contact_request_decline</a>, id: 702850 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/1dc8a8aaa39640bab120e5ac22bfb517?auth=6dbe1d2861d2387590fd9f50e25bad29#64">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/1eac35043abc46caa7523ec2d5c2eeaa?auth=83d3f0440e70e7fd55157c8b896a2f75#58">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3670697">test_add_contact_field_validation</a>, id: 702777 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/1dc8a8aaa39640bab120e5ac22bfb517?auth=6dbe1d2861d2387590fd9f50e25bad29#142">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/1eac35043abc46caa7523ec2d5c2eeaa?auth=83d3f0440e70e7fd55157c8b896a2f75#196">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestActivityMultipleDevicePR:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3670720">test_activity_center_reply_read_unread_delete_filter_swipe</a>, id: 702947 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/df05b442bd0a4e1b9a006a962a5b771e?auth=49d46ebfa7eea42ae703253b737f4b96#129">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/e114aa8fa2f44b5caf1ec944a0c81e5c?auth=86fc42e22137b3a892528d0aba2387ed#101">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestActivityMultipleDevicePRTwo:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3670723">test_activity_center_admin_notification_accept_swipe</a>, id: 702958 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/2f7edb8dbff84ff2a9f7c9de5d9c0cf2?auth=d2c4eeead2b082b217aa3232bdf1a20e#166">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/2cb72c79576b4a29b0609b7df0dfae46?auth=176f72ade576a49a3ddfb406e6b82931#119">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3670722">test_activity_center_mentions</a>, id: 702957 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/2f7edb8dbff84ff2a9f7c9de5d9c0cf2?auth=d2c4eeead2b082b217aa3232bdf1a20e#125">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/2cb72c79576b4a29b0609b7df0dfae46?auth=176f72ade576a49a3ddfb406e6b82931#103">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestWalletMultipleDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3670738">test_wallet_send_asset_from_drawer</a>, id: 727230 </b></td></tr><tr><td></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3670737">test_wallet_send_eth</a>, id: 727229 </b></td></tr><tr><td></td></tr></tbody></table><h4>Class TestDeepLinksOneDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3670735">test_links_open_universal_links_from_chat</a>, id: 704613 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/203a395665c6464da2803d5cf442d303?auth=bae99f678497a6b631c142857610e559#62">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3670696">test_links_deep_links_profile</a>, id: 702775 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/203a395665c6464da2803d5cf442d303?auth=bae99f678497a6b631c142857610e559#125">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestOneToOneChatMultipleSharedDevicesNewUi:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3670716">test_1_1_chat_edit_message</a>, id: 702855 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/03879485b264426dbc8e31108876b860?auth=e9107d9b382827d80e6dd4668e4821a6#460">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/4ad85886a069434b95d3528a611204de?auth=19cc880d1acc4b415594325f370bf84e#298">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3670690">test_1_1_chat_message_reaction</a>, id: 702730 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/03879485b264426dbc8e31108876b860?auth=e9107d9b382827d80e6dd4668e4821a6#97">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/4ad85886a069434b95d3528a611204de?auth=19cc880d1acc4b415594325f370bf84e#92">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3670695">test_1_1_chat_non_latin_messages_stack_update_profile_photo</a>, id: 702745 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/03879485b264426dbc8e31108876b860?auth=e9107d9b382827d80e6dd4668e4821a6#353">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/4ad85886a069434b95d3528a611204de?auth=19cc880d1acc4b415594325f370bf84e#217">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3670691">test_1_1_chat_pin_messages</a>, id: 702731 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/03879485b264426dbc8e31108876b860?auth=e9107d9b382827d80e6dd4668e4821a6#215">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/4ad85886a069434b95d3528a611204de?auth=19cc880d1acc4b415594325f370bf84e#172">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>5. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3670693">test_1_1_chat_text_message_delete_push_disappear</a>, id: 702733 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/03879485b264426dbc8e31108876b860?auth=e9107d9b382827d80e6dd4668e4821a6#552">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/4ad85886a069434b95d3528a611204de?auth=19cc880d1acc4b415594325f370bf84e#404">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>6. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3670705">test_1_1_chat_push_emoji</a>, id: 702813 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/03879485b264426dbc8e31108876b860?auth=e9107d9b382827d80e6dd4668e4821a6#411">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/4ad85886a069434b95d3528a611204de?auth=19cc880d1acc4b415594325f370bf84e#267">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>7. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3670698">test_1_1_chat_emoji_send_reply_and_open_link</a>, id: 702782 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/03879485b264426dbc8e31108876b860?auth=e9107d9b382827d80e6dd4668e4821a6#149">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/4ad85886a069434b95d3528a611204de?auth=19cc880d1acc4b415594325f370bf84e#127">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>8. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3670730">test_1_1_chat_send_image_save_and_share</a>, id: 703391 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/03879485b264426dbc8e31108876b860?auth=e9107d9b382827d80e6dd4668e4821a6#491">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/4ad85886a069434b95d3528a611204de?auth=19cc880d1acc4b415594325f370bf84e#344">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestOneToOneChatMultipleSharedDevicesNewUiTwo:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3670699">test_1_1_chat_is_shown_message_sent_delivered_from_offline</a>, id: 702783 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/7925e95f2ee2499f8c70304dfd450127?auth=5fa9a9d673016d08aadd3ad08f88edab#100">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/1c3936beddc645a990fa9667cd68d7cd?auth=ef35f8feaa71fbe53a0250c3b6ec4b6f#92">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3670700">test_1_1_chat_delete_via_long_press_relogin</a>, id: 702784 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/7925e95f2ee2499f8c70304dfd450127?auth=5fa9a9d673016d08aadd3ad08f88edab#179">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/1c3936beddc645a990fa9667cd68d7cd?auth=ef35f8feaa71fbe53a0250c3b6ec4b6f#169">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestCommunityOneDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3670694">test_community_copy_and_paste_message_in_chat_input</a>, id: 702742 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/a5cfc79f0f774c08a767a924fc53a2ad?auth=c35684077c227ef2b10a956c213a5f83#78">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3670713">test_community_navigate_to_channel_when_relaunch</a>, id: 702846 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/a5cfc79f0f774c08a767a924fc53a2ad?auth=c35684077c227ef2b10a956c213a5f83#54">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3670725">test_restore_multiaccount_with_waku_backup_remove_switch</a>, id: 703133 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/a5cfc79f0f774c08a767a924fc53a2ad?auth=c35684077c227ef2b10a956c213a5f83#231">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3670718">test_community_undo_delete_message</a>, id: 702869 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/a5cfc79f0f774c08a767a924fc53a2ad?auth=c35684077c227ef2b10a956c213a5f83#107">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>5. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3670729">test_community_mute_community_and_channel</a>, id: 703382 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/a5cfc79f0f774c08a767a924fc53a2ad?auth=c35684077c227ef2b10a956c213a5f83#126">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>6. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3670733">test_community_discovery</a>, id: 703503 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/a5cfc79f0f774c08a767a924fc53a2ad?auth=c35684077c227ef2b10a956c213a5f83#356">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table></details>

--- a/src/status_im/contexts/communities/events.cljs
+++ b/src/status_im/contexts/communities/events.cljs
@@ -2,6 +2,7 @@
   (:require
     [clojure.string :as string]
     [legacy.status-im.data-store.communities :as data-store.communities]
+    [oops.core :as oops]
     [schema.core :as schema]
     [status-im.constants :as constants]
     [status-im.contexts.chat.messenger.messages.link-preview.events :as link-preview.events]
@@ -25,7 +26,7 @@
                   token-permissions-check id last-opened-at]
            :as   community}       (data-store.communities/<-rpc community-js)
           previous-last-opened-at (get-in db [:communities id :last-opened-at])]
-      (when (>= clock (get-in db [:communities id :clock]))
+      (when (and id (>= clock (get-in db [:communities id :clock])))
         {:db (assoc-in db
               [:communities id]
               (assoc community :last-opened-at (max last-opened-at previous-last-opened-at)))
@@ -223,14 +224,14 @@
       :fx [[:dispatch [:communities/check-permissions-to-join-community community-id]]]})))
 
 (defn community-fetched
-  [{:keys [db]} [community-id community]]
-  (when community
+  [{:keys [db]} [community-id community-js]]
+  (when community-js
     {:db (update db :communities/fetching-communities dissoc community-id)
-     :fx [[:dispatch [:communities/handle-community community]]
+     :fx [[:dispatch [:communities/handle-community community-js]]
           [:dispatch [:chat.ui/spectate-community community-id]]
           [:dispatch
            [:chat.ui/cache-link-preview-data (link-preview.events/community-link community-id)
-            community]]]}))
+            (data-store.communities/<-rpc community-js)]]]}))
 
 (rf/reg-event-fx :chat.ui/community-fetched community-fetched)
 
@@ -253,18 +254,19 @@
              (not (get-in db [:communities community-id]))
              (not (get-in db [:communities/fetching-communities community-id])))
     {:db            (assoc-in db [:communities/fetching-communities community-id] true)
-     :json-rpc/call [{:method     "wakuext_fetchCommunity"
-                      :params     [{:CommunityKey    community-id
-                                    :TryDatabase     true
-                                    :WaitForResponse true}]
-                      :on-success (fn [community]
-                                    (if community
-                                      (rf/dispatch [:chat.ui/community-fetched community-id
-                                                    community])
-                                      (failed-to-fetch-community
-                                       community-id
-                                       "community wasn't found at the store node")))
-                      :on-error   (partial failed-to-fetch-community community-id)}]}))
+     :json-rpc/call [{:method      "wakuext_fetchCommunity"
+                      :params      [{:CommunityKey    community-id
+                                     :TryDatabase     true
+                                     :WaitForResponse true}]
+                      :js-response true
+                      :on-success  (fn [community]
+                                     (if community
+                                       (rf/dispatch [:chat.ui/community-fetched community-id
+                                                     community])
+                                       (failed-to-fetch-community
+                                        community-id
+                                        "community wasn't found at the store node")))
+                      :on-error    (partial failed-to-fetch-community community-id)}]}))
 
 (schema/=> fetch-community
   [:=>
@@ -283,11 +285,15 @@
 (rf/reg-event-fx :communities/fetch-community fetch-community)
 
 (defn spectate-community-success
-  [{:keys [db]} [{:keys [communities]}]]
-  (when-let [community (first communities)]
-    {:db (-> db
-             (assoc-in [:communities (:id community) :spectated] true))
-     :fx [[:dispatch [:communities/handle-community community]]]}))
+  [{:keys [db]} [response-js]]
+  (when response-js
+    (let [communities  (oops/oget response-js "communities")
+          community    (first communities)
+          community-id (when community (oops/oget community "id"))]
+      (when community
+        {:db (-> db
+                 (assoc-in [:communities community-id :spectated] true))
+         :fx [[:dispatch [:communities/handle-community community]]]}))))
 
 (rf/reg-event-fx :chat.ui/spectate-community-success spectate-community-success)
 
@@ -295,13 +301,14 @@
   [{:keys [db]} [community-id]]
   (let [{:keys [spectated joined]} (get-in db [:communities community-id])]
     (when (and (not joined) (not spectated))
-      {:json-rpc/call [{:method     "wakuext_spectateCommunity"
-                        :params     [community-id]
-                        :on-success [:chat.ui/spectate-community-success]
-                        :on-error   (fn [err]
-                                      (log/error {:message
-                                                  "Failed to spectate community"
-                                                  :error err}))}]})))
+      {:json-rpc/call [{:method      "wakuext_spectateCommunity"
+                        :params      [community-id]
+                        :js-response true
+                        :on-success  [:chat.ui/spectate-community-success]
+                        :on-error    (fn [err]
+                                       (log/error {:message
+                                                   "Failed to spectate community"
+                                                   :error err}))}]})))
 
 (schema/=> spectate-community
   [:=>

--- a/src/status_im/contexts/communities/events_test.cljs
+++ b/src/status_im/contexts/communities/events_test.cljs
@@ -1,5 +1,6 @@
 (ns status-im.contexts.communities.events-test
   (:require [cljs.test :refer [deftest is testing]]
+            [legacy.status-im.data-store.communities :as data-store.communities]
             matcher-combinators.test
             [status-im.contexts.chat.messenger.messages.link-preview.events :as link-preview.events]
             [status-im.contexts.communities.events :as events]))
@@ -51,7 +52,7 @@
                      [:dispatch [:chat.ui/spectate-community community-id]]
                      [:dispatch
                       [:chat.ui/cache-link-preview-data "community-link+community-id"
-                       {:id community-id}]]]}
+                       (data-store.communities/<-rpc #js {"id" community-id})]]]}
                (events/community-fetched cofx arg))))))
     (testing "given a joined community"
       (let [cofx {:db {:communities/fetching-communities {community-id true}}}
@@ -62,7 +63,7 @@
                      [:dispatch [:chat.ui/spectate-community community-id]]
                      [:dispatch
                       [:chat.ui/cache-link-preview-data "community-link+community-id"
-                       {:id community-id}]]]}
+                       (data-store.communities/<-rpc #js {"id" community-id})]]]}
                (events/community-fetched cofx arg))))))
     (testing "given a token-gated community"
       (let [cofx {:db {:communities/fetching-communities {community-id true}}}
@@ -73,7 +74,7 @@
                      [:dispatch [:chat.ui/spectate-community community-id]]
                      [:dispatch
                       [:chat.ui/cache-link-preview-data "community-link+community-id"
-                       {:id community-id}]]]}
+                       (data-store.communities/<-rpc #js {"id" community-id})]]]}
                (events/community-fetched cofx arg))))))
     (testing "given nil community"
       (testing "do nothing"

--- a/src/status_im/contexts/communities/events_test.cljs
+++ b/src/status_im/contexts/communities/events_test.cljs
@@ -104,26 +104,38 @@
            (events/spectate-community {:db {:communities {community-id {}}}} [community-id]))))))
 
 (deftest spectate-community-success-test
-  (testing "given communities"
-    (testing "mark first community spectated true"
-      (is (match?
-           {:db {:communities {community-id {:spectated true}}}}
-           (events/spectate-community-success {} [{:communities [{:id community-id}]}]))))
-    (testing "dispatch fxs for first community"
-      (is (match?
-           {:fx [[:dispatch [:communities/handle-community {:id community-id}]]
-                ]}
-           (events/spectate-community-success {} [{:communities [{:id community-id}]}])))))
-  (testing "given empty community"
-    (testing "do nothing"
-      (is (match?
-           nil
-           (events/spectate-community-success {} [{:communities []}])))))
-  (testing "given nil community"
-    (testing "do nothing"
-      (is (match?
-           nil
-           (events/spectate-community-success {} []))))))
+  (let [community-id-1 1
+        community-id-2 2]
+    (testing "given communities"
+      (testing "mark first community spectated true"
+        (is (match?
+             {:db {:communities {community-id-1 {:spectated true}}}}
+             (events/spectate-community-success {}
+                                                (clj->js [{:communities [{:id community-id-1}
+                                                                         {:id community-id-2}]}])))))
+      (testing "dispatch fxs for first community"
+        ;; Convert the JavaScript object back to a Clojure map because `match?` does not work well
+        ;; with JavaScript objects.
+        (is (match?
+             {:fx [[:dispatch [:communities/handle-community {:id community-id-1}]]]}
+             (let [result (events/spectate-community-success {}
+                                                             (clj->js [{:communities
+                                                                        [{:id community-id-1}
+                                                                         {:id community-id-2}]}]))]
+               (update-in result [:fx 0 1] #(js->clj % :keywordize-keys true)))))))
+    (testing "given empty community"
+      (testing "do nothing"
+        (is (match?
+             nil
+             (events/spectate-community-success {}
+                                                (clj->js [{:communities []}]))))))
+    (testing "given nil community"
+      (testing "do nothing"
+        (is (match?
+             nil
+             (events/spectate-community-success {}
+                                                (clj->js []))))))))
+
 
 (deftest get-revealed-accounts-test
   (let [community {:id community-id}]

--- a/src/utils/transforms.cljs
+++ b/src/utils/transforms.cljs
@@ -18,7 +18,7 @@
 (defn <-js-map
   "Shallowly transforms JS Object keys/values with `key-fn`/`val-fn`.
 
-  Returns nil if `m` is not an instance of `js/Object`.
+  Returns m if `m` is not an instance of `js/Object`.
 
   Implementation taken from `js->clj`, but with the ability to customize how
   keys and/or values are transformed in one loop.
@@ -33,7 +33,8 @@
   ([^js m]
    (<-js-map m nil))
   ([^js m {:keys [key-fn val-fn]}]
-   (when (identical? (type m) js/Object)
+   (assert (or (nil? m) (identical? (type m) js/Object) (map? m)))
+   (if (identical? (type m) js/Object)
      (persistent!
       (reduce (fn [r k]
                 (let [v       (oops/oget+ m k)
@@ -41,7 +42,8 @@
                       new-val (if val-fn (val-fn k v) v)]
                   (assoc! r new-key new-val)))
               (transient {})
-              (js-keys m))))))
+              (js-keys m)))
+     m)))
 
 (defn js-stringify
   [js-object spaces]

--- a/src/utils/transforms_test.cljs
+++ b/src/utils/transforms_test.cljs
@@ -13,7 +13,7 @@
     (are [expected m]
      (is (equals-as-json expected (sut/<-js-map m)))
      nil               nil
-     nil               #js []
+     {:a 1}            {:a 1}
      #js {}            #js {}
      #js {"a" 1 "b" 2} #js {"a" 1 "b" 2}))
 

